sys/kern/link_elf.c

```diff
@@ -468,7 +493,17 @@ link_elf_init(void* arg)
 		ef->modptr = modptr;
 		baseptr = preload_search_info(modptr, MODINFO_ADDR);
 		if (baseptr != NULL)
+#ifndef __CHERI_PURE_CAPABILITY__
 			linker_kernel_file->address = *(caddr_t *)baseptr;
+#else
+			/*
+			 * MODINFO_ADDR is really a virtual address,
+			 * not a pointer
+			 */
+			linker_kernel_file->address = cheri_setoffset(
+			    linker_kernel_file->address,
+			    *(vm_offset_t *)baseptr);
```

Comment:
cheri_setaddress? Though this provenance is a bit weird, but, the whole function is a bit dodgy for that.

Reply:
It is a bit weird.  Since the base is 0, the offset vs address is ambiguous.  Probably it should be setaddress.